File: project/app/src/main/java/org/owntracks/android/support/OpenCageResponse.java
package org.owntracks.android.support;

import android.support.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.util.List;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class OpenCageResponse {
    private List<OpenCageResult> results;


    public List<OpenCageResult> getResults() {
        return results;
    }

    public @Nullable String getFormatted() {
        return results.size() > 0 ? results.get(0).getFormatted() : null;
    }
}

File: project/app/src/main/java/org/owntracks/android/support/OpenCageResponse.kt
package org.owntracks.android.support

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
internal class OpenCageResponse {
    val results: List<OpenCageResult>? = null
    val formatted: String?
        get() = if (results != null && results.isNotEmpty()) results[0].formatted else null
}